### PR TITLE
fix(e2e): ensure we check if there are deployments incase pods are no…

### PR DESCRIPTION
**Pull request by aslakknutsen**

### CODE REVIEW

File: e2e/infra/pod_state.go [hunk 1]
```diff
@@ -7,6 +7,56 @@ import (
 	"github.com/maistra/istio-workspace/test/shell"
 )
 
+// AllDeploymentsAndPodsReady checks if both AllDeploymentsReady(Deployment) and AllPodsReady return true
```
Comment (bartoszmajsak) [inline e2e/infra/pod_state.go:10]:
```suggestion
// AllDeploymentsAndPodsReady checks if both AllDeploymentsReady(Deployment) and AllPodsReady return true.
```

File: e2e/infra/pod_state.go [hunk 2]
```diff
@@ -7,6 +7,56 @@ import (
...
+func AllDeploymentsAndPodsReady(ns string) func() bool {
+	return func() bool {
+		return AllDeploymentsReady("deployment", ns)() && AllPodsReady(ns)()
+	}
+}
+
+// AllDeploymentConfigsAndPodsReady checks if both AllDeploymentsReady(DeploymentConfig) and AllPodsReady return true
```
Comment (bartoszmajsak) [inline e2e/infra/pod_state.go:17]:
```suggestion
// AllDeploymentConfigsAndPodsReady checks if both AllDeploymentsReady(DeploymentConfig) and AllPodsReady return true.
```

**Review by bartoszmajsak — APPROVED** [review]
Minor `godot` issues as suggestions. Otherwise - lets ship it!

**Comment by alien-ike** [comment]
@aslakknutsen: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-tests-openshift | ffa211dbcad26fe5175d74b6171a73c1334f0ffb | [link](http://status-prow.apps.workspace.maistra.test/view/gs/ike-prow-logs/pr-logs/pull/maistra_istio-workspace/614/e2e-tests-openshift/1351149877765083136) | `/test openshift`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by alien-ike** [comment]
@aslakknutsen: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-tests-openshift | 56a31e74b3274e5de5ec966a0708d2a8e70a1b0c | [link](http://status-prow.apps.workspace.maistra.test/view/gs/ike-prow-logs/pr-logs/pull/maistra_istio-workspace/614/e2e-tests-openshift/1351177812735692800) | `/test openshift`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by alien-ike** [comment]
@aslakknutsen: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-tests-openshift | 03d5b115789b90eabe0532f9dc2c3af94f3f6782 | [link](http://status-prow.apps.workspace.maistra.test/view/gs/ike-prow-logs/pr-logs/pull/maistra_istio-workspace/614/e2e-tests-openshift/1351194546968465408) | `/test openshift`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by alien-ike** [comment]
@aslakknutsen: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-tests-openshift | 4cddb5c00097777805d110ff35d4cb6078404c4a | [link](http://status-prow.apps.workspace.maistra.test/view/gs/ike-prow-logs/pr-logs/pull/maistra_istio-workspace/614/e2e-tests-openshift/1351211660521508864) | `/test openshift`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by alien-ike** [comment]
@aslakknutsen: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-tests-openshift | 24145ec66096c785bccc2917ca1893cca1c035d2 | [link](http://status-prow.apps.workspace.maistra.test/view/gs/ike-prow-logs/pr-logs/pull/maistra_istio-workspace/614/e2e-tests-openshift/1351463695439368192) | `/test openshift`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by alien-ike** [comment]
@aslakknutsen: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-tests-openshift | 2dfc972168a9d9d5f39cdbcd3601bbcb307060ac | [link](http://status-prow.apps.workspace.maistra.test/view/gs/ike-prow-logs/pr-logs/pull/maistra_istio-workspace/614/e2e-tests-openshift/1351477033196064768) | `/test openshift`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by alien-ike** [comment]
@aslakknutsen: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-tests-openshift | dc1a71cede20c58ec3efac169ccc0bd6cf4c7bd4 | [link](http://status-prow.apps.workspace.maistra.test/view/gs/ike-prow-logs/pr-logs/pull/maistra_istio-workspace/614/e2e-tests-openshift/1351639973241032704) | `/test openshift`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Pull request merged**